### PR TITLE
ChromHMM enhancer extension

### PR DESCRIPTION
**Pull request by LeonHafner**
This pull request introduces the Enhancer Prediction with ChromHMM, as described in Issue #63.
The changes have not yet been tested and are not yet complete.

**TODO:**
- [x] Add better way to create the cellmarkfiletable using the input csv for ehmm
- [x] Add java container for BINARIZE_BAMS and LEARN_MODEL processes
- [x] Get parameters from the config file (organism, genome version, ...)
- [x] Add nf-core meta object to the input and output channels of the processes

### CODE REVIEW

File: modules/local/ehmm/construct_model.nf
Comment (nictru) [inline modules/local/ehmm/construct_model.nf:1]: This change should not be part of this PR

File: subworkflows/local/chromhmm.nf
Comment (nictru) [inline subworkflows/local/chromhmm.nf:1]: Make sure indent in main is consistent

**Review by nictru — CHANGES_REQUESTED** [review]
Copy-pasting files from another repo is not a good practice, we should do one of the following:
- Reference files from there as Permalinks to the [original repo](https://github.com/jernst98/ChromHMM) directly in the igenomes config (preferable)
- Add the repo as git submodule

Resulting TODO list:
- [ ] Remove chromHMM files from this Repo
- [ ] Add meta entries to all processes, will contain metadata from the design file (state, assay) as well as a unique id (which should be added as tag to each process)
- [x] Add labels to each process (important for resource allocation)
- [x] Add java docker images: Even if it runs on our cluster, it does not have to run on every user's system
- I thought of something else and forgot it, will at later if it comes to my mind again

File: modules/local/ehmm/construct_model.nf
Comment (LeonHafner) [reply modules/local/ehmm/construct_model.nf:1]: Just a note: We leave this commit in this PR, otherwise it will be missing in the merged version.

**Comment by LeonHafner** [comment]
I integrated the ChromHMM workflow into the main pipeline and testet it.
This part should be done from my side and can be merged.
Feel free to make any necessary corrections before approving the merge.

**Review by nictru — CHANGES_REQUESTED** [review]
Implementation looks fine, but the ChromHMM output is not yet used in any way

**Comment by LeonHafner** [comment]
I integrated the output of ChromHMM/Rose into the pipeline. This should finish the work on ChromHMM/Rose. Feel free to merge the PR or request further changes if needed.

File: conf/modules.config [hunk 1]
```diff
@@ -52,12 +52,22 @@ process {
...
+        ext.suffix = "gff"
+    }
+
     withName: ROSE_OUTPUT_TO_BED {
         ext.args = {"'BEGIN{FS=\"\\t\";OFS=\"\\t\"} {print \$1, \$4-1, \$5}'"}
         ext.prefix = {"enhancers_${meta.state}"}
         ext.suffix = "bed"
     }
 
+    withName: SUBTRACT_BLACKLIST {
```
Comment (LeonHafner) [inline conf/modules.config:67]: Had to add a config for this module due to a file name conflict

**Review by nictru — APPROVED** [review]

**Pull request merged**